### PR TITLE
feat: Text display of param values

### PR DESCRIPTION
**Pull request by UberWaffe**
Future PR for displaying actions and conditions as strings in the event edit window will follow on this.
This is only for displaying parameter values as text in the action and condition edit windows.

### CODE REVIEW

File: src/core/lang.c [hunk 1]
```diff
@@ -551,6 +551,40 @@ const uint8_t *lang_get_string(int group, int index)
...
+        case BUILDING_HOUSE_SMALL_PALACE:
+        case BUILDING_HOUSE_MEDIUM_PALACE:
+        case BUILDING_HOUSE_LARGE_PALACE:
+        case BUILDING_HOUSE_LUXURY_PALACE:
+        case BUILDING_NATIVE_HUT:
+        case BUILDING_NATIVE_MEETING:
+        case BUILDING_NATIVE_CROPS:
+            return lang_get_string(41, type);
+        default:
+            break;
```
Comment (crudelios) [inline src/core/lang.c:582]:
```suggestion
    if (building_is_house(type) || type == BUILDING_NATIVE_HUT ||
        type == BUILDING_NATIVE_MEETING || type == BUILDING_NATIVE_CROPS) {
        return lang_get_string(41, type);
    } else {
        return lang_get_string(28, type);
    }
```

File: src/scenario/scenario_events_parameter_data.c [hunk 2]
```diff
@@ -1,5 +1,10 @@
 #include "scenario_events_parameter_data.h"
+#include "core/lang.h"
```
Comment (crudelios) [inline src/scenario/scenario_events_parameter_data.c:2]: Don't forget to add an empty line between the main include and the other ones.

File: src/translation/english.c [hunk 3]
```diff
@@ -1130,6 +1131,19 @@ static translation_string all_strings[] = {
...
+    {TR_PARAMETER_VALUE_BUILDING_SMALL_TEMPLE_MERCURY, "Small Mercury temple"},
+    {TR_PARAMETER_VALUE_BUILDING_SMALL_TEMPLE_MARS, "Small Mars temple"},
+    {TR_PARAMETER_VALUE_BUILDING_SMALL_TEMPLE_VENUS, "Small Venus temple"},
+    {TR_PARAMETER_VALUE_BUILDING_LARGE_TEMPLE_CERES, "Large Ceres temple"},
+    {TR_PARAMETER_VALUE_BUILDING_LARGE_TEMPLE_NEPTUNE, "Large Neptune temple"},
+    {TR_PARAMETER_VALUE_BUILDING_LARGE_TEMPLE_MERCURY, "Large Mercury temple"},
+    {TR_PARAMETER_VALUE_BUILDING_LARGE_TEMPLE_MARS, "Large Mars temple"},
+    {TR_PARAMETER_VALUE_BUILDING_LARGE_TEMPLE_VENUS, "Large Venus temple"},
+    {TR_PARAMETER_VALUE_BUILDING_SENATE_UPGRADED, "Senate upgraded"},
+    {TR_PARAMETER_VALUE_BUILDING_FORUM_UPGRADED, "Forum upgraded"},
```
Comment (crudelios) [inline src/translation/english.c:1146]: Upgraded senate and forum don't really exist, so I don't think you need these strings.

File: src/translation/translation.h [hunk 4]
```diff
@@ -1124,6 +1125,19 @@ typedef enum {
...
+    TR_PARAMETER_VALUE_BUILDING_SMALL_TEMPLE_MERCURY,
+    TR_PARAMETER_VALUE_BUILDING_SMALL_TEMPLE_MARS,
+    TR_PARAMETER_VALUE_BUILDING_SMALL_TEMPLE_VENUS,
+    TR_PARAMETER_VALUE_BUILDING_LARGE_TEMPLE_CERES,
+    TR_PARAMETER_VALUE_BUILDING_LARGE_TEMPLE_NEPTUNE,
+    TR_PARAMETER_VALUE_BUILDING_LARGE_TEMPLE_MERCURY, 
+    TR_PARAMETER_VALUE_BUILDING_LARGE_TEMPLE_MARS,
+    TR_PARAMETER_VALUE_BUILDING_LARGE_TEMPLE_VENUS,
+    TR_PARAMETER_VALUE_BUILDING_SENATE_UPGRADED,
+    TR_PARAMETER_VALUE_BUILDING_FORUM_UPGRADED,
```
Comment (crudelios) [inline src/translation/translation.h:1140]: Same here.

File: src/window/editor/scenario_action_edit.c [hunk 5]
```diff
@@ -55,10 +56,19 @@ static struct {
     int parameter_being_edited;
     int parameter_being_edited_current_value;
 
+    uint8_t display_text[MAX_TEXT_LENGTH];
+
     scenario_action_t *action;
     scenario_action_data_t *xml_info;
 } data;
 
+static uint8_t *translation_for_paramvalue(parameter_type type, int value)
```
Comment (crudelios) [inline src/window/editor/scenario_action_edit.c:65]:
```suggestion
static uint8_t *translation_for_param_value(parameter_type type, int value)
```

File: src/window/editor/scenario_condition_edit.c [hunk 6]
```diff
@@ -55,10 +56,19 @@ static struct {
     int parameter_being_edited;
     int parameter_being_edited_current_value;
 
+    uint8_t display_text[MAX_TEXT_LENGTH];
+
     scenario_condition_t *condition;
     scenario_condition_data_t *xml_info;
 } data;
 
+static uint8_t *translation_for_paramvalue(parameter_type type, int value)
```
Comment (crudelios) [inline src/window/editor/scenario_condition_edit.c:65]:
```suggestion
static uint8_t *translation_for_param_value(parameter_type type, int value)
```

**Review by crudelios — COMMENTED** [review]
Some small changes.

File: src/core/lang.c (hunk 1 above)
Comment (UberWaffe) [reply src/core/lang.c:582]: Done.

File: src/translation/english.c (hunk 3 above)
Comment (UberWaffe) [reply src/translation/english.c:1146]: Done.

File: src/translation/translation.h (hunk 4 above)
Comment (UberWaffe) [reply src/translation/translation.h:1140]: Done.

File: src/window/editor/scenario_condition_edit.c (hunk 6 above)
Comment (UberWaffe) [reply src/window/editor/scenario_condition_edit.c:65]: Done.

**Comment by UberWaffe** [comment]
@crudelios Changes made.

**Comment by UberWaffe** [comment]
@crudelios I swear I am a proper software engineer.
Pushed now.

File: src/core/lang.c [hunk 7]
```diff
@@ -1,6 +1,7 @@
-#include "building/type.h"
 #include "core/lang.h"
 
+#include "building/building.h"
+#include "building/type.h"
```
Comment (crudelios) [inline src/core/lang.c:4]: No need to include `building/type.h`. It's already included on `building/building.h`.

**Review by crudelios — COMMENTED** [review]
A very small nitpick.

File: src/core/lang.c (hunk 7 above)
Comment (UberWaffe) [reply src/core/lang.c]: Fixed.

**Comment by UberWaffe** [comment]
Oh, I see there are conflicts now.
If you are happy with the change, then I'll do a rebase.

**Review by crudelios — APPROVED** [review]

**Comment by crudelios** [comment]
Ah, you need to fix the merge conflicts now @UberWaffe

**Comment by UberWaffe** [comment]
@crudelios Allright, rebased and conflicts fixed.

**Pull request merged**